### DISCUSSION TITLE
Thermostat controls with modulation via ebusd

### DISCUSSION THREAD

**Discussion by JasperTack** [john30/ebusd#707] [Q&A]
I have the following setup for heating my home: I have a Vaillant boiler with thermostat VRC700 with a control unit VR70. This thermostat is not placed inside the heated rooms, but in a separate technical room. Which makes it not possible to use the builtin temperature sensor of the thermostat to regulate the heating.

My question now is: would it be possible to use the ebus port to send out the actual temperature of the rooms (coming from my knx sensors in each room) to the installation. Is there anybody who has attempted such a thing yet?
Another way might be to have my own thermostat implementation which runs on my raspberry pi and send out the instructions to the ebus port like any other thermostat would do.

Thanks!

**Comment by namelessone667** [comment]
Hi! @JasperTack  did you managed to wourk this out? I wish to use a similar setup, thanks!

**Comment by namelessone667** [comment]
in VRC700 controller configuration (https://github.com/john30/ebusd-configuration/blob/master/ebusd-2.1.x/en/vaillant/15.700.csv) there is a variable called z1RoomTemp, I think that is the room temperature, but in the configuration it is marked as read only. I am new to ebuds, can someone more experienced suggest if this variable can be chagned to writable and write the value periodically through ebusd to emulate a room thermostat?

**Reply by cyberthom42** [reply]
No, it is not possible to write this value. What exactly are you trying to achieve? If enabled the room temperature usually is only used to adapt the heat curve. However, I do not know your boiler.

**Reply by namelessone667** [reply]
I would like to emulate a room thermostat (or more specificaly, a room temperature sensor). The VRC700 controller is located on the heat pump, so the internal temperature sensor is not usable. This room temperature is needed for cooling regulation ( I have a heat pump with cooling capabilities)

**Reply by rICTx-T1D** [reply]
@john30 hi John,
what is necessary to emulate an ebus thermometer?

 In professional menu it is possible to join it to one heating circuit. In my case vrc720 is mapped to hc1. I have the same foul, because my installer has crews it on heating hydraulic station. My technician room is 4-5°C hotter than the rest of house.

**Reply by JonesGW-MGD** [reply]
Prefacing this by stating the obvious that many, including myself, will tell you that this isn't the intention of ebusd so you need to accept the risks - known and unknown but you wouldn't be the first to want to do this and others have been successful to some extent.

I'm pretty sure you can amend RoomTemp or z1RoomTemp messages (depending on your controller) to be writable.  There are many messages that are by default set to read-only because of the obvious damage that can be done.  You should try searching for 'replace thermostat' or 'emulate controller' on the discussion forum.

You will likely need an automation that regularly interacts with the heat source and publishes the required temperature even if the required temp is unchanged to avoid contention with hardware controller or the heat/cooling source resetting periodically.  Again, people have posted about this.

Goes without saying you would need to test extensively and understand what the outcomes might be if your automation platform failed.

For that reason most people would recommend against this.  This may feel like an unpopular suggestion but would it not be better to purchase a VRC720F (wireless controller)?   I know everything Vaillant can be eye-wateringly expensive but there are used examples of this device on a popular auction site for less than half retail price - I guess it depends how much you value your time but you could be doing an awful lot of hours experimenting with a DIY solution.

**Comment by Elijen** [comment]
I wanted to do something similar, but Vaillant only supports few zones (just one in my case) and I have many rooms that each get different amount of sun and require different heating.

The better way in my opinion is to manually control flow temperatures (indirectly by setting required room temperature between 5-30 degrees) and using valves on your manifold/radiators that communicate with your temperature sensors.

**Reply by JonesGW-MGD** [reply]
The 'Vailliant solution' to this is to use VR50 AmbiSENSE valves heads (TRVs) on each emitter - they are able to directly 'call for heat'  and differ therefore from most 'smart TRVs' which rely on a central thermostat(s) to call for heat and the Smart TRV simply sets the flow/temp to each radiator providing that a 'call for heat' is in play - the 'smart' bit being you can access each TRV via an App (except perhaps the Honeywell evoHome which I understand can individually call for heat but I would never suggest non-Vaillant controls on a Vaillant system).   If you have VR50s, it makes the temp control set on the VRC7xx redundant, instead it displays the 'Required Temperature' being the highest value of any of the VR50s.  (That is to say, the heat source will 'fire' anytime the set-point of any TRV is higher than its actual temp.  And remember, the temp sense on the VR50 is centimetres from the emitter so does not represent room temp - you need to adjust your settings accordingly by experimentation therefore).

Doing this, you end up with a zone per room (In fact I've set mine up so that I have a zone per TRV - so my rooms that have two radiators are technically two zones.)

People criticise the MyVaillant app but it works 'OK' and essentially gives you remote access to all your zones.  There is also a PyVaillant HA automation(which I haven't used so cannot vouch for)  that can now apparently integrate the VR50s into HA.  (The VR50s do not use ebus - instead they are RF linked to the VR920/1 Internet hub.)

Of course, since it's the Vaillant solution, it's also pretty expensive !!- (I've got 13 VR50s, and two range extenders so I know!).
